dev-packages/node-integration-tests/suites/tracing/postgresjs/test.ts

```diff
@@ -215,11 +409,239 @@ describe('postgresjs auto instrumentation', () => {
+            'db.system.name': 'postgres',
+            'db.operation.name': 'SELECT',
+            'db.query.text': `SELECT * FROM "User" WHERE "email" = '${EXISTING_TEST_EMAIL}'`,
+            'custom.requestHook': 'called',
+            'sentry.op': 'db',
+            'sentry.origin': 'auto.db.postgresjs',
+            'server.address': 'localhost',
+            'server.port': 5444,
+          }),
+          description: `SELECT * FROM "User" WHERE "email" = '${EXISTING_TEST_EMAIL}'`,
+          op: 'db',
+          status: 'ok',
+          origin: 'auto.db.postgresjs',
+        }),
+      ]),
+      extra: expect.objectContaining({
+        requestHookCalled: expect.objectContaining({
+          database: 'test_db',
+          host: 'localhost',
+          port: '5444',
+          sanitizedQuery: expect.any(String),
+        }),
+      }),
+    };
+
+    await createRunner(__dirname, 'scenario-requestHook.mjs')
+      .withFlags('--import', `${__dirname}/instrument-requestHook.mjs`)
+      .withDockerCompose({ workingDirectory: [__dirname], readyMatches: ['port 5432'] })
+      .expect({ transaction: EXPECTED_TRANSACTION })
+      .start()
+      .completed();
+  });
+
+  // Tests for URL-based initialization pattern (regression prevention)
+  test('should instrument postgres package with URL initialization (CJS)', { timeout: 90_000 }, async () => {
+    const EXPECTED_TRANSACTION = {
+      transaction: 'Test Transaction',
+      spans: expect.arrayContaining([
+        createDbSpanMatcher('CREATE TABLE'),
+        createDbSpanMatcher('INSERT'),
+        createDbSpanMatcher('UPDATE'),
+        createDbSpanMatcher('SELECT'),
+      ]),
+    };
+
+    await createRunner(__dirname, 'scenario-url.cjs')
+      .withDockerCompose({ workingDirectory: [__dirname], readyMatches: ['port 5432'] })
+      .expect({ transaction: EXPECTED_TRANSACTION })
+      .start()
+      .completed();
+  });
+
+  test('should instrument postgres package with URL initialization (ESM)', { timeout: 90_000 }, async () => {
+    const EXPECTED_TRANSACTION = {
+      transaction: 'Test Transaction',
+      spans: expect.arrayContaining([
+        createDbSpanMatcher('CREATE TABLE'),
+        createDbSpanMatcher('INSERT'),
+        createDbSpanMatcher('SELECT'),
+        createDbSpanMatcher('DELETE'),
+      ]),
+    };
+
+    await createRunner(__dirname, 'scenario-url.mjs')
+      .withFlags('--import', `${__dirname}/instrument.mjs`)
+      .withDockerCompose({ workingDirectory: [__dirname], readyMatches: ['port 5432'] })
+      .expect({ transaction: EXPECTED_TRANSACTION })
+      .start()
+      .completed();
+  });
+
+  test('should instrument sql.unsafe() queries (CJS)', { timeout: 90_000 }, async () => {
+    const EXPECTED_TRANSACTION = {
+      transaction: 'Test Transaction',
+      spans: expect.arrayContaining([
+        createDbSpanMatcher('CREATE TABLE'),
+        createDbSpanMatcher('INSERT'),
+        createDbSpanMatcher('SELECT'),
```

Comment:
what's the query here? Can we test the span description and the `db.query.text` attribute?

Reply:
Updated the tests to check all properties: https://github.com/getsentry/sentry-javascript/pull/17961/commits/e687a7911d86b1fdaf34abc1f5c9b96a7e696b04